src/EPR.SubsidiaryBulkUpload.Application/Extensions/UserRequestModelExtensions.cs

```diff
@@ -0,0 +1,49 @@
+﻿using EPR.SubsidiaryBulkUpload.Application.Models;
+
+namespace EPR.SubsidiaryBulkUpload.Application.Extensions;
+
+public static class UserRequestModelExtensions
+{
+    public static UserRequestModel ToUserRequestModel(this IDictionary<string, string> metadata)
+    {
+        if (metadata is null)
+        {
+            return null;
+        }
+
+        var caseInsensitiveMetadata = new Dictionary<string, string>(metadata, StringComparer.OrdinalIgnoreCase);
+
+        if (!caseInsensitiveMetadata.ContainsKey("UserId") || !caseInsensitiveMetadata.ContainsKey("OrganisationId"))
```

Comment:
Sonarqube issue: Prefer a 'TryGetValue' call over a Dictionary indexer access guarded by a 'ContainsKey' check to avoid double lookup

Both ContainsKey checks here can be replaced with TryGetValue with output to vars so you don't need to look up again later

Reply:
Updated.